spec/system/articles/user_discussion_locks_spec.rb

```diff
@@ -29,6 +33,37 @@
       visit article.path
       expect(page).to have_selector("#new_comment")
     end
+
+    it "doesn't hide reply button on each comment on the Article page" do
+      visit article.path
+      expect(page).to have_selector("[data-tracking-name='comment_reply_button']").twice
```

Comment:
**Nitpick/non-blocking suggestion:** I wish we could avoid using the `data-tracking-name` attribute in these tests. I'm not sure if this attribute will stay in the codebase for the long term, but that's a personal preference/opinion of mine.

If someone else feels strongly about this I think an alternative that might work here would be to add a `data-testid` to the reply element. We use this pattern in certain areas of the app for these types of scenarios, [for example here](https://github.com/forem/forem/blob/main/app/javascript/article-form/components/ArticleCoverImage.jsx#L49)

Reply:
@aitchiss might have some thoughts; I know a lot of Cypress tests are around the `aria-label` and `aria-role`, that might make sense for this as well.

Reply:
I think we should steer away from the tracking name, as @fdocr mentions it's very likely to be removed in the near future. I think using a `data-testid` for these kind of specs would be fine, but it would be nice to also add to the existing Cypress E2E tests so that we check the integrated user flow too.

@thejwuscript how do you feel about adding Cypress specs? (we have existing ones that could be added to at `cypress/integration/seededFlows/articleFlows/manageFlows/lockDiscussion.spec.js`). If you'd need assistance with that, just let me know.

Reply:
I've changed the system specs to use `data-testid` instead and added some cypress specs. 
The specs are very similar to what I wrote for the system specs.